clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1008,12 +1011,11 @@ public long committed(TopicPartition partition) {
         try {
             Cluster cluster = this.metadata.fetch();
             List<PartitionInfo> parts = cluster.partitionsForTopic(topic);
-            if (parts == null) {
-                metadata.add(topic);
-                client.awaitMetadataUpdate();
-                parts = metadata.fetch().partitionsForTopic(topic);
-            }
-            return parts;
+            if (parts != null)
+                return parts;
+
+            Map<String, List<PartitionInfo>> topicMetadata = fetcher.getTopicMetadata(Arrays.asList(topic), requestTimeoutMs);
```

Comment:
Since we depend on the metadata matching the subscription set in this patch, I wanted to remove any other calls which can affect it. The alternative would be to always intersect the subscription and metadata before computing the hash sent along in the join group. Either way should work, but I actually think it's a good thing not having the persistent metadata state updated by partitionsFor(). Can you explain a little more why this is less than ideal?